libdnf/rpm/solv/goal_private.hpp

```diff
@@ -152,11 +150,13 @@ inline void GoalPrivate::set_installonly(const std::vector<std::string> & instal
     }
 }
 
-inline void GoalPrivate::add_install(IdQueue & queue, bool strict) {
+inline void GoalPrivate::add_install(IdQueue & queue, bool strict, bool best) {
     // TODO dnf_sack_make_provides_ready(sack); When provides recomputed job musy be empty
     Id what = pool_queuetowhatprovides(pool, &queue.get_queue());
     staging.push_back(
-        SOLVER_INSTALL | SOLVER_SOLVABLE_ONE_OF | SOLVER_SETARCH | SOLVER_SETEVR | (strict ? 0 : SOLVER_WEAK), what);
+        SOLVER_INSTALL | SOLVER_SOLVABLE_ONE_OF | SOLVER_SETARCH | SOLVER_SETEVR | (strict ? 0 : SOLVER_WEAK) |
+            (best ? 0 : SOLVER_FORCEBEST),
```

Comment:
Really confusing. At first glance, it seems the opposite. `best ? SOLVER_FORCEBEST : 0`. I see it on 3 places. I'm confused or it is bug? I see it in 3 places. Am I confused or is it a bug?

Reply:
Thanks, it is a bug.